core/gui/src/app/workspace/component/menu/menu.component.ts

```diff
@@ -203,21 +276,77 @@ export class MenuComponent implements OnInit, OnDestroy {
     disable: boolean;
     onClick: () => void;
   } {
+    // If workflow is invalid, always disable and show "Invalid Workflow"
+    if (!this.isWorkflowValid) {
+      return {
+        text: "Invalid Workflow",
+        icon: "warning",
+        disable: true,
+        onClick: () => {},
+      };
+    }
+
+    // If workflow is empty, always disable and show "Empty Workflow"
     if (this.isWorkflowEmpty) {
       return {
-        text: "Empty",
-        icon: "exclamation-circle",
+        text: "Empty Workflow",
+        icon: "info-circle",
+        disable: true,
+        onClick: () => {},
+      };
+    }
+
+    // Check for creating/connecting states
+    if (
+      this.computingUnitStatusService.isCreatingUnitValue ||
+      this.computingUnitStatusService.isConnectingToUnitValue
+    ) {
+      return {
+        text: "Connecting",
+        icon: "loading",
         disable: true,
         onClick: () => {},
       };
-    } else if (!this.isWorkflowValid) {
+    }
+
+    // This handles the case where a unit exists but we're not connected to it
+    if (
+      environment.computingUnitManagerEnabled &&
+      this.computingUnitStatus !== ComputingUnitConnectionState.NoComputingUnit &&
+      !this.computingUnitConnected
+    ) {
       return {
-        text: "Error",
-        icon: "exclamation-circle",
+        text: "Connecting",
+        icon: "loading",
         disable: true,
         onClick: () => {},
       };
     }
+
+    // In cuManager mode with no computing unit, show "Create Compute" button
+    if (
+      environment.computingUnitManagerEnabled &&
+      this.computingUnitStatus === ComputingUnitConnectionState.NoComputingUnit
+    ) {
+      return {
+        text: "Create Compute",
```

Comment:
Change it to `Connect` and make the button shorter.

Reply:
to avoid confusing with another `connect` case, pop a confirmation window saying a computing unit will be created with default cpu and memory setting.